File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityEquipmentPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.inventory.item.ItemStack;
import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record EntityEquipmentPacket(int entityId, short slot, ItemStack itemStack)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeShort(slot);
        writeSlot(os, itemStack);
    }

    @Override
    public byte getId() {
        return 0x04;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityEquipmentPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.inventory.item.ItemStack
import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityEquipmentPacket(val entityId: Int, val slot: Short, val itemStack: ItemStack) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeShort(slot.toInt())
        os.writeSlot(itemStack)
    }

    override val id: Byte
        get() = 0x04
}
